test/externalTests/trade.js

```diff
@@ -1,27 +1,50 @@
-// const assert = require('assert')
+const assert = require('assert')
 
 module.exports = () => (bot, done) => {
-  done()
-  // TODO fix unreliable trading/trading test
-  /*
   const mcData = require('minecraft-data')(bot.version)
   const Item = require('prismarine-item')(bot.version)
 
   const villagerType = mcData.entitiesByName.villager ? 'villager' : 'Villager'
 
+  const summonCommand = bot.supportFeature('indexesVillagerRecipes')
+    ? `/summon ${villagerType} ~ ~1 ~ {NoAI:1, Offers:{Recipes:[0:{maxUses:7,buy:{id:"minecraft:emerald",Count:2},sell:{id:"minecraft:wheat",Count:2}, uses:1}]}}`
```

Comment:
I don't know how it works in minecraft but is `{Recipes:[0:{}]}` correct?
(Just a question)

Reply:
I dont know much either, but I've been testing it and it works!

Here's a standard villager metadata in 1.8 for reference:
```
{HurtByTimestamp:0,Attributes:[0:{Base:20.0d,Name:"generic.maxHealth"},1:{Base:0.0d,Name:"generic.knockbackResistance"},2:{Base:0.5d,Name:"generic.movementSpeed"},3:{Base:16.0d,Modifiers:[0:{UUIDMost:-3631253411971314151L,UUIDLeast:-8828509508598644614L,Amount:-0.01844724002497035d,Operation:1,Name:"Random spawn bonus"}],Name:"generic.followRange"}],Riches:0,Invulnerable:0b,ForcedAge:0,PortalCooldown:0,AbsorptionAmount:0.0f,FallDistance:0.0f,DeathTime:0s,DropChances:[0:0.085f,1:0.085f,2:0.085f,3:0.085f,4:0.085f],PersistenceRequired:0b,HealF:20.0f,Age:0,Motion:[0:0.0d,1:-0.0784000015258789d,2:0.0d],Leashed:0b,UUIDLeast:-7028543208059295380L,Health:20s,Air:300s,OnGround:1b,Dimension:0,Offers:{Recipes:[0:{maxUses:7,buy:{id:"minecraft:leather",Count:11b,Damage:0s},sell:{id:"minecraft:emerald",Count:1b,Damage:0s},uses:0,rewardExp:1b},1:{maxUses:7,buy:{id:"minecraft:emerald",Count:3b,Damage:0s},sell:{id:"minecraft:leather_leggings",Count:1b,Damage:0s},uses:0,rewardExp:1b}]},Rotation:[0:0.0f,1:0.0f],Profession:4,UUIDMost:6991824022769256519L,Equipment:[0:{},1:{},2:{},3:{},4:{}],Pos:[0:-26.0987337365875d,1:4.0d,2:20.005468739674903d],Fire:-1s,CanPickUpLoot:1b,HurtTime:0s,CareerLevel:1,Career:2,Inventory:[],Willing:0b}
```

Reply:
weird.